File: packages/types/src/enums.ts
export enum UserRole {
  BUYER = 'BUYER',
  SELLER = 'SELLER',
  ADMIN = 'ADMIN',
  SUPPORT = 'SUPPORT',
}

export enum StoreStatus {
  PENDING_KYC = 'PENDING_KYC',
  ACTIVE = 'ACTIVE',
  SUSPENDED = 'SUSPENDED',
  REJECTED = 'REJECTED',
}

export enum SellerType {
  FIRST_PARTY = 'FIRST_PARTY', // Waw direct retail / inventory
  THIRD_PARTY = 'THIRD_PARTY', // Marketplace vendor
}

export enum PaymentMethod {
  RAAST_P2M_QR = 'RAAST_P2M_QR', // State Bank of Pakistan Raast Instant QR (via PostEx XPay)
  XPAY_CARD = 'XPAY_CARD', // Visa, Mastercard, PayPak (via PostEx XPay)
  XPAY_WALLET_JAZZCASH = 'XPAY_WALLET_JAZZCASH', // JazzCash Mobile Account (via PostEx XPay)
  XPAY_WALLET_EASYPAISA = 'XPAY_WALLET_EASYPAISA', // Easypaisa Mobile Wallet (via PostEx XPay)
  COD = 'COD', // Cash on Delivery (PostEx Courier Collection)
}

export enum PaymentStatus {
  PENDING = 'PENDING',
  AUTHORIZED = 'AUTHORIZED',
  PAID = 'PAID',
  ESCROW_HELD = 'ESCROW_HELD', // SBP Escrow holding period
  COD_PENDING = 'COD_PENDING',
  COD_COLLECTED = 'COD_COLLECTED',
  FAILED = 'FAILED',
  REFUNDED = 'REFUNDED',
}

export enum OrderStatus {
  PENDING = 'PENDING',
  CONFIRMED = 'CONFIRMED',
  PROCESSING = 'PROCESSING',
  SHIPPED = 'SHIPPED',
  OUT_FOR_DELIVERY = 'OUT_FOR_DELIVERY',
  DELIVERED = 'DELIVERED',
  CANCELLED = 'CANCELLED',
  RETURN_REQUESTED = 'RETURN_REQUESTED',
  RETURNED = 'RETURNED',
}

export enum CourierProvider {
  POSTEX = 'POSTEX', // Designated primary logistics & COD partner
  LEOPARDS = 'LEOPARDS',
  TRAX = 'TRAX',
  WAW_FLEET = 'WAW_FLEET',
}

export enum PayoutStatus {
  SCHEDULED = 'SCHEDULED',
  PROCESSING = 'PROCESSING',
  COMPLETED = 'COMPLETED',
  ON_HOLD = 'ON_HOLD',
}

export enum ReturnStatus {
  REQUESTED = 'REQUESTED',
  APPROVED = 'APPROVED',
  PICKUP_SCHEDULED = 'PICKUP_SCHEDULED',
  IN_TRANSIT = 'IN_TRANSIT',
  INSPECTED = 'INSPECTED',
  REFUNDED = 'REFUNDED',
  REJECTED = 'REJECTED',
}

export enum ReturnReason {
  DAMAGED_OR_DEFECTIVE = 'DAMAGED_OR_DEFECTIVE',
  SIZE_OR_FIT_MISMATCH = 'SIZE_OR_FIT_MISMATCH',
  ITEM_NOT_AS_DESCRIBED = 'ITEM_NOT_AS_DESCRIBED',
  CHANGED_MIND = 'CHANGED_MIND',
}

File: apps/mobile/lib/core/enums/enums.dart
enum UserRole { buyer, seller, admin }

enum OrderStatus {
  pending,
  confirmed,
  processing,
  shipped,
  outForDelivery,
  delivered,
  cancelled,
  returnRequested,
  returned,
}

enum PaymentMethod {
  cod,
  alfaWallet,
  alfalahAccount,
  alfaCard,
  raastQr,
}

enum PaymentStatus {
  pending,
  authorized,
  paid,
  escrowHeld,
  codPending,
  codCollected,
  failed,
  refunded,
}

enum StoreStatus {
  pendingKyc,
  active,
  suspended,
  rejected,
}

enum SellerType {
  firstParty,
  thirdParty,
}

enum PayoutStatus {
  pending,
  processing,
  completed,
  failed,
}

extension OrderStatusExtension on OrderStatus {
  String get label {
    switch (this) {
      case OrderStatus.pending:
        return 'Pending';
      case OrderStatus.confirmed:
        return 'Confirmed';
      case OrderStatus.processing:
        return 'Processing';
      case OrderStatus.shipped:
        return 'Shipped';
      case OrderStatus.outForDelivery:
        return 'Out for Delivery';
      case OrderStatus.delivered:
        return 'Delivered';
      case OrderStatus.cancelled:
        return 'Cancelled';
      case OrderStatus.returnRequested:
        return 'Return Requested';
      case OrderStatus.returned:
        return 'Returned';
    }
  }
}

extension PaymentMethodExtension on PaymentMethod {
  String get label {
    switch (this) {
      case PaymentMethod.cod:
        return 'Cash on Delivery';
      case PaymentMethod.alfaWallet:
        return 'Alfa Wallet';
      case PaymentMethod.alfalahAccount:
        return 'Alfalah Bank Account';
      case PaymentMethod.alfaCard:
        return 'Credit/Debit Card';
      case PaymentMethod.raastQr:
        return 'RAAST QR';
    }
  }

  String get icon {
    switch (this) {
      case PaymentMethod.cod:
        return '💵';
      case PaymentMethod.alfaWallet:
        return '📱';
      case PaymentMethod.alfalahAccount:
        return '🏦';
      case PaymentMethod.alfaCard:
        return '💳';
      case PaymentMethod.raastQr:
        return '📱';
    }
  }
}

PaymentMethod parsePaymentMethod(String? value) {
  switch (value) {
    case 'COD':
      return PaymentMethod.cod;
    case 'ALFA_WALLET':
      return PaymentMethod.alfaWallet;
    case 'ALFALAH_ACCOUNT':
      return PaymentMethod.alfalahAccount;
    case 'ALFA_CARD':
      return PaymentMethod.alfaCard;
    case 'RAAST_P2M_QR':
      return PaymentMethod.raastQr;
    default:
      return PaymentMethod.cod;
  }
}

OrderStatus parseOrderStatus(String? value) {
  switch (value) {
    case 'PENDING':
      return OrderStatus.pending;
    case 'CONFIRMED':
      return OrderStatus.confirmed;
    case 'PROCESSING':
      return OrderStatus.processing;
    case 'SHIPPED':
      return OrderStatus.shipped;
    case 'OUT_FOR_DELIVERY':
      return OrderStatus.outForDelivery;
    case 'DELIVERED':
      return OrderStatus.delivered;
    case 'CANCELLED':
      return OrderStatus.cancelled;
    case 'RETURN_REQUESTED':
      return OrderStatus.returnRequested;
    case 'RETURNED':
      return OrderStatus.returned;
    default:
      return OrderStatus.pending;
  }
}
